### PR TITLE
docs: add more warnings for default secrets and docker-compose

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY

Add more warnings on our default for docker-compose and stress out the importance of having a unique random `SECRET_KEY`

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: docs/docs/installation/configuring-superset.mdx [hunk 1]
```diff
@@ -99,6 +99,10 @@ SECRET_KEY = 'YOUR_OWN_RANDOM_GENERATED_SECRET_KEY'
 
 You can generate a strong secure key with `openssl rand -base64 42`.
 
+> Your secret key will be used for securely signing session cookies
```
Comment (sfirke) [inline docs/docs/installation/configuring-superset.mdx:102]:
```suggestion
:::caution Your secret key will be used for securely signing session cookies
```

File: docs/docs/installation/configuring-superset.mdx [hunk 2]
```diff
@@ -99,6 +99,10 @@ SECRET_KEY = 'YOUR_OWN_RANDOM_GENERATED_SECRET_KEY'
...
+  and encrypting sensitive information on the database
```
Comment (sfirke) [inline docs/docs/installation/configuring-superset.mdx:103]:
```suggestion
and encrypting sensitive information stored in Superset's application metadata database.
```

File: docs/docs/installation/configuring-superset.mdx [hunk 3]
```diff
@@ -99,6 +99,10 @@ SECRET_KEY = 'YOUR_OWN_RANDOM_GENERATED_SECRET_KEY'
...
+  Make sure you are changing this key for your deployment with a strong key.
```
Comment (sfirke) [inline docs/docs/installation/configuring-superset.mdx:104]:
```suggestion
Ensure your deployment uses a strong, unique key.
Store this key securely. Without this key, you will be unable to access your Superset metadata database. That is, you will lose all of your Superset dashboards, charts, etc. :::
```

**Review by sfirke — APPROVED** [review]
@dpgaspar I left a suggestion to change from quote format to caution format (you can see an example of how that renders on the [docker compose install page](https://superset.apache.org/docs/installation/installing-superset-using-docker-compose/)).  And I put a note in there about keeping your key, what do you think about that?  People have written in Slack about getting locked out of their Superset instance when they don't know their SECRET_KEY that they randomly generated.  Maybe they store it as an env variable and then delete a file or wipe out a VM or something.

**Comment by mistercrunch** [comment]
One thing I was thinking about recently around `ENVIRONMENT_TAG_CONFIG` and `SUPERSET_ENV` is we could either force a very visible tag or refuse to start when use a bad combination of things (default `SECRET_KEY` and "SUPERSET_ENV=production").

Also was thinking that in-your-face, "things you should know" in dev would be useful, notices like "hey we noticed you're on python 3.9 - the supported version is now 3.10, you should upgrade"

File: docs/docs/installation/configuring-superset.mdx (hunk 3 above)
Comment (csummers) [reply docs/docs/installation/configuring-superset.mdx:104]: This `:::caution` block as merged is not closed with a `:::`, so this page renders everything after like so:

image

The suggestion did have a closing `:::`, though.

Comment (sfirke) [reply docs/docs/installation/configuring-superset.mdx:104]: Thanks @csummers! I have a PR open now to fix this.

**Pull request merged**